src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -446,6 +447,56 @@ public void CleanupAfterEachTest()
+        /// </summary>
+        /// <param name="databaseType">database type.</param>
+        /// <param name="providedConnectionString">connection string provided in the config.</param>
+        /// <param name="expectedUpdatedConnectionString">Updated connection string with Application Name.</param>
+        /// <param name="isHostedScenario">If Dab is hosted or OSS.</param>
+        [DataTestMethod]
+        [DataRow(DatabaseType.MSSQL, "Something;", "Something;Application Name=dab_oss_1.0.0;", false, DisplayName = "[MSSQL]:Connection String doesn't contain Application name for DAB oss.")]
+        [DataRow(DatabaseType.MySQL, "Something;", "Something;", false, DisplayName = "[MYSQL]:Connection String doesn't contain Application name for DAB oss.")]
+        [DataRow(DatabaseType.PostgreSQL, "Something;", "Something;", false, DisplayName = "[PGSQL]:Connection String doesn't contain Application name for DAB oss.")]
+        [DataRow(DatabaseType.CosmosDB_PostgreSQL, "Something;", "Something;", false, DisplayName = "[COSMOSDB_PGSQL]:Connection String doesn't contain Application name for DAB oss.")]
+        [DataRow(DatabaseType.CosmosDB_NoSQL, "Something;", "Something;", false, DisplayName = "[COSMOSDB_NOSQL]:Connection String doesn't contain Application name for DAB oss.")]
+        [DataRow(DatabaseType.MSSQL, "Something;Application Name=CustAppName;", "Something;Application Name=CustAppName,dab_oss_1.0.0;", false, DisplayName = "[MSSQL]:Connection String contains customer Application name for DAB oss.")]
+        [DataRow(DatabaseType.MSSQL, "Something1;Application Name=CustAppName;Something2;", "Something1;Application Name=CustAppName,dab_oss_1.0.0;Something2;", false, DisplayName = "[MSSQL2]:Connection String contains customer Application name for DAB oss.")]
+        [DataRow(DatabaseType.MySQL, "Something;Application Name=CustAppName;", "Something;Application Name=CustAppName;", false, DisplayName = "[MYSQL]:Connection String contains customer Application name for DAB oss.")]
+        [DataRow(DatabaseType.PostgreSQL, "Something;Application Name=CustAppName;", "Something;Application Name=CustAppName;", false, DisplayName = "[PGSQL]:Connection String contains customer Application name for DAB oss.")]
+        [DataRow(DatabaseType.CosmosDB_PostgreSQL, "Something;Application Name=CustAppName;", "Something;Application Name=CustAppName;", false, DisplayName = "[COSMOSDB_PGSQL]:Connection String contains customer Application name for DAB oss.")]
+        [DataRow(DatabaseType.CosmosDB_NoSQL, "Something;Application Name=CustAppName;", "Something;Application Name=CustAppName;", false, DisplayName = "[COSMOSDB_NOSQL]:Connection String contains customer Application name for DAB oss.")]
+        [DataRow(DatabaseType.MSSQL, "Something;", "Something;Application Name=dab_hosted_1.0.0;", true, DisplayName = "[MSSQL]:Connection String doesn't contain Application name for DAB hosted.")]
```

Comment:
```suggestion
        [DataRow(DatabaseType.MSSQL, "Something;", "Something;Application Name=dab_hosted_1.0.0;", true, DisplayName = "[MSSQL]:Connection String contains Application name for DAB hosted.")]
```

Reply:
it's correct, right? the connection string doesn't contain so we add Application name property. I'll update the description of testcases.